loaders/src/glTF/2.0/Extensions/ExtrasAsMetadata.ts

```diff
@@ -0,0 +1,104 @@
+import { Nullable } from "babylonjs/types";
+import { TransformNode } from "babylonjs/Meshes/transformNode";
+import { Camera } from "babylonjs/Cameras/camera";
+
+import { IProperty } from "babylonjs-gltf2interface";
+import { INode, ICamera, IMaterial } from "../glTFLoaderInterfaces";
+import { IGLTFLoaderExtension } from "../glTFLoaderExtension";
+import { GLTFLoader } from "../glTFLoader";
+import { Material } from "babylonjs/Materials/material";
+
+const NAME = "ExtrasAsMetadata";
+
+interface ObjectWithMetadata {
+    metadata: any;
+}
+
+/**
+ * Store glTF extras (if present) in BJS objects' metadata
+ */
+export class ExtrasAsMetadata implements IGLTFLoaderExtension {
+    /** The name of this extension. */
+    public readonly name = NAME;
+
+    /** Defines whether this extension is enabled. */
+    public enabled = true;
+
+    private _loader: GLTFLoader;
+
+    private _assignExtras(
+        babylonObject: ObjectWithMetadata,
+        gltfProp: IProperty
+    ): void {
+        if (gltfProp.extras && Object.keys(gltfProp.extras).length > 0) {
+            if (!babylonObject.metadata) {
+                babylonObject.metadata = {};
+            }
+            Object.assign(babylonObject.metadata, gltfProp.extras);

```

Comment:
I forgot about this before, but the loader currently already writes a `metadata.gltf.pointers` to created Babylon objects.

https://github.com/BabylonJS/Babylon.js/blob/e22385628759e47b1eb89bf7fc66e4a48f95930a/loaders/src/glTF/2.0/glTFLoader.ts#L1983

It might be good to put the extras metadata at `metadata.gltf.extras` so that it's far less likely to conflict with other metadata. It will also make it clearer what extras to write when exporting a glTF.

Reply:
Done